File: rb/features.rb
# CrisisCoreFusion SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module CrisisCoreFusionFeatures
  def self.make_feature(name)
    case name
    when "base"
      CrisisCoreFusionBaseFeature.new
    when "test"
      CrisisCoreFusionTestFeature.new
    else
      CrisisCoreFusionBaseFeature.new
    end
  end
end

File: py/features.py
# CrisisCoreFusion SDK feature factory

from feature.base_feature import CrisisCoreFusionBaseFeature
from feature.test_feature import CrisisCoreFusionTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CrisisCoreFusionBaseFeature(),
        "test": lambda: CrisisCoreFusionTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
